### PR TITLE
Attach directly to event names instead of prefixes

### PR DESCRIPTION
**Pull request by arkgil**
This PR changes the semantics of attaching handlers as proposed in #1. `attach/4` now accepts an event name and the handler will be invoked only when this particular event has been emitted. Additionally I've added an `attach_many/4` function which allows to attach a single handler to many events. 

/cc @studzien

### CODE REVIEW

**Comment by arkgil** [comment]
> should we remove the Agent implementation

Yes, I think we can remove it, the ETS one should be the baseline shall we introduce some optimizations.

> In my opinion, we could get rid of the Impl indirection altogether

Let's keep it for a little more 😄

File: lib/telemetry/impl/ets.ex [hunk 1]
```diff
@@ -69,29 +65,14 @@ defmodule Telemetry.Impl.Ets do
   @spec handler_exists?(Telemetry.handler_id()) :: boolean()
   defp handler_exists?(handler_id) do
     case :ets.match(@table, {handler_id, :_, :_, :_, :_}) do
-      [_] ->
+      [_ | _] ->
```
Comment (studzien) [inline lib/telemetry/impl/ets.ex]: Is it actually possible to have more than one element returned here?

Comment (arkgil) [reply lib/telemetry/impl/ets.ex]: Yes, when we `attach_many` there is an object in ETS table for each event the handler is attached to. Each of these handlers have the same ID.

**Comment by arkgil** [comment]
I guess it's good to merge :)

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**